p_soc_auto_base/utils.py

```diff
@@ -39,6 +42,32 @@
 LOGGER = logging.getLogger('django')
 
 
+def show_milliseconds(seconds):
+    """
+    formats the input to a string with milliseconds if said input is less
+    than 1 second, otherwise the input is formatted to a string with seconds
+
+    :arg seconds: an input object that represents a duration in a format that
+        can be coerced to :class:`decimal.Decimal`
+
+    :returns: soemthing like '1.200 seconds' or '525 milliseconds'
+
+    :raises: :exc:`TypeError` if the input cannot be coerced to
+        :class:`decimal.Decimal`
+
+    """
+    try:
+        seconds = decimal.Decimal(seconds)
```

Comment:
We need a float, so why are we converting to Decimal?
Are we afraid `float(seconds) > 1` will fail somehow?